main/lsp/requests/rename.cc

```diff
@@ -111,67 +114,80 @@ unique_ptr<WorkspaceEdit> Renamer::buildEdit() {
     we->documentChanges = move(textDocEdits);
     return we;
 }
+
 class MethodRenamer : public Renamer {
 public:
-    MethodRenamer(const string oldName, const string newName) : Renamer(oldName, newName) {}
+    MethodRenamer(const core::GlobalState &gs, const LSPConfiguration &config, const string oldName,
+                  const string newName)
+        : Renamer(gs, config, oldName, newName) {}
     ~MethodRenamer() {}
-    void rename(const core::GlobalState &gs, unique_ptr<Location> &location, core::FileRef fref) override {
-        auto loc = location->range->toLoc(gs, fref);
+    void rename(unique_ptr<core::lsp::QueryResponse> &response) override {
+        auto loc = response->getLoc();
         auto source = loc.source(gs);
+        auto location = config.loc2Location(gs, loc);
         string newsrc;
-        if (absl::StartsWith(source, "def ") || absl::StartsWith(source, "attr_reader") ||
-            absl::StartsWith(source, "attr_writer") || absl::StartsWith(source, "attr_accessor")) {
+        if (auto sendResp = response->isSend()) {
+            auto receiverLoc = sendResp->receiverLoc;
+            ENFORCE(loc.contains(receiverLoc), "receiver expression not within send expression");
+            ENFORCE(loc.beginPos() == receiverLoc.beginPos(), "send expression doesn't start with receiver");
+            string::size_type methodNameOffset;
+            if (receiverLoc.endPos() == receiverLoc.beginPos()) {
+                // no receiver expression, method is start of loc
+                methodNameOffset = 0;
+            } else {
+                // find the method name: <receiver expression> <whitespace?> <dot> <whitespace?> <method name>
```

Comment:
This seems to be the same logic as in `prepare_rename.cc`. Should this be a helper method on `SendResponse`? (getMethodNameLoc()?) Then you could potentially even unit test this logic!